File: backend/app/models/project.py
from sqlalchemy import Column, String, Integer, ForeignKey, DateTime, Text
from sqlalchemy.orm import relationship
import datetime
from ..database import Base
import uuid


class Project(Base):
    __tablename__ = "projects"
    
    id = Column(String, primary_key=True)
    name = Column(String)
    description = Column(Text, nullable=True)
    status = Column(String)  # inPlans, inProgress, onPause, completed
    priority = Column(String)  # high, medium, low
    assignee = Column(String)
    order = Column(Integer)
    created_at = Column(DateTime, default=datetime.datetime.utcnow)
    
    # Foreign keys
    owner_id = Column(String, ForeignKey("users.id"))
    dashboard_id = Column(String, ForeignKey("dashboards.id", ondelete="CASCADE"), nullable=True)
    
    # Relationships
    owner = relationship("User", back_populates="projects")
    dashboard = relationship("Dashboard", back_populates="projects")
    task_columns = relationship("TaskColumn", back_populates="project", cascade="all, delete-orphan")

File: newbackend/src/main/java/tech/nicorp/pm/projects/domain/Project.java
package tech.nicorp.pm.projects.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import tech.nicorp.pm.dashboards.domain.Dashboard;
import tech.nicorp.pm.users.domain.User;

import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "projects")
@Getter
@Setter
@NoArgsConstructor
public class Project {
    @Id
    @Column(columnDefinition = "uuid")
    private UUID id = UUID.randomUUID();

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "description", columnDefinition = "text")
    private String description;

    @Column(name = "status", nullable = false)
    private String status;

    @Column(name = "priority", nullable = false)
    private String priority;

    @Column(name = "assignee")
    private String assignee;

    @Column(name = "order_index")
    private Integer orderIndex;

    @Column(name = "color")
    private String color;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "dashboard_id")
    private Dashboard dashboard;

    @Column(name = "created_at", nullable = false)
    private OffsetDateTime createdAt = OffsetDateTime.now();
}



